### PR TITLE
Add Bedrock Embedding

### PR DESCRIPTION
**Pull request by viveksilimkhan1**
# Description

Add Bedrock Embedding 

Fixes # (#8549)

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/embeddings/bedrock.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+class Models(str):
+    TITAN_EMBEDDING = "amazon.titan-embed-text-v1"
+
+
+# To be used when AWS Bedrock comes with more embedding models
+# class ModelParams(str):
+# TITAN_EMBEDDING = {"inputText":""}
+
+
+class BedrockEmbeddings(BaseEmbedding):
```
Comment (Disiok) [inline llama_index/embeddings/bedrock.py:20]: can we rename to `BedrockEmbedding` for naming consistency?

File: llama_index/embeddings/bedrock.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    _client: Any = PrivateAttr()
+
+    def __init__(
+        self,
+        model_name: str = Models.TITAN_EMBEDDING,
+        max_length: int = 8000,
```
Comment (Disiok) [inline llama_index/embeddings/bedrock.py:26]: I don't see this being used anywhere?

File: llama_index/embeddings/bedrock.py (hunk 1 above)
Comment (viveksilimkhan1) [reply llama_index/embeddings/bedrock.py:20]: Yes, definitely, will do it.

File: llama_index/embeddings/bedrock.py (hunk 2 above)
Comment (viveksilimkhan1) [reply llama_index/embeddings/bedrock.py:26]: Currently max_length is not being used as the query is truncated automatically once the length crosses it's limit (i.e. 8000). I'll set a warning for query crossing the limit right away.

Comment (viveksilimkhan1) [reply llama_index/embeddings/bedrock.py:26]: I wanted to set warning to check if the number of tokens exceed the limit, however boto3 handles this, hence I am removing it.

Comment (viveksilimkhan1) [reply llama_index/embeddings/bedrock.py]: Hi @Disiok , I have made the changes, please review.

**Comment by Disiok** [comment]
Thanks! Made some minor fixes, going to land this after CI passes!

**Comment by viveksilimkhan1** [comment]
Thanks for reviewing @Disiok

**Comment by kandakji** [comment]
Hi, I've done some tests, the assert statements in the code can prevent IAM Role or Instance Profile usage when hosting the app within AWS without using Access keys.

**Pull request merged**